doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -384,6 +452,22 @@ guess the missing piece of information.
 Let-in definitions
 ------------------
 
+.. insertgram term_let name_list_comma
+
+.. productionlist:: coq
+   term_let : let `name` `binder_list_opt` `colon_term_opt` := `term` in `term`
+            : let `single_fix` in `term`
+            : let `names_tuple` `return_type_opt` := `term` in `term`
+            : let ' `pattern` := `term` in `term`
+            : let ' `pattern` := `term` return `term100` in `term`
```

Comment:
Let's create a `return_type` nonterminal that corresponds to ```return `term100` ```, use it everywhere it can be used, and define `return_type_opt` relatively to `return_type` (and maybe we have to change the name `return_type_opt` as well so that it is clear that it is more than `return_type` or `empty`.